src/ctapipe/tools/compute_irf.py

```diff
@@ -0,0 +1,637 @@
+            self.bias_resolution_maker(
+                events=self.signal_events[self.signal_events["selected"]],
+            )
+        )
+        hdus.append(
+            self.angular_resolution_maker(
+                events=self.signal_events[self.signal_events["selected_gh"]],
+            )
+        )
+        if self.do_background:
+            if self.opt_result.spatial_selection_table is None:
+                raise ValueError(
+                    "Calculating the point-source sensitivity requires "
+                    f"theta cuts, but {self.cuts_file} does not contain any."
+                )
+
+            hdus.append(
+                self.sensitivity_maker(
+                    signal_events=self.signal_events[self.signal_events["selected"]],
+                    background_events=self.background_events[
+                        self.background_events["selected_gh"]
+                    ],
+                    spatial_selection_table=self.opt_result.spatial_selection_table,
+                    gamma_spectrum=self.gamma_target_spectrum,
+                )
+            )
+        return hdus
+
+    def start(self):
+        """
+        Load events and calculate the irf (and the benchmarks).
+        """
+        reduced_events = dict()
+        for sel in self.particles:
+            if sel.epp.gammaness_classifier != self.opt_result.clf_prefix:
+                raise RuntimeError(
+                    "G/H cuts are only valid for gammaness scores predicted by "
+                    "the same classifier model. Requested model: %s. "
+                    "Model used for g/h cuts: %s."
+                    % (
+                        sel.epp.gammaness_classifier,
+                        self.opt_result.clf_prefix,
+                    )
+                )
+
+            if (
+                sel.epp.quality_query.quality_criteria
+                != self.opt_result.precuts.quality_criteria
+            ):
+                self.log.warning(
+                    "Precuts are different from precuts used for calculating "
+                    "g/h / theta cuts. Provided precuts:\n%s. "
+                    "\nUsing the same precuts as g/h / theta cuts:\n%s. "
+                    % (
+                        sel.epp.quality_query.to_table(functions=True)[
+                            "criteria", "func"
+                        ],
+                        self.opt_result.precuts.to_table(functions=True)[
+                            "criteria", "func"
+                        ],
+                    )
+                )
+                sel.epp.quality_query = EventQualityQuery(
+                    parent=sel,
+                    quality_criteria=self.opt_result.precuts.quality_criteria,
+                )
+
+            self.log.debug(
+                "%s Precuts: %s" % (sel.kind, sel.epp.quality_query.quality_criteria)
+            )
+            evs, cnt, meta = sel.load_preselected_events(self.chunk_size, self.obs_time)
```

Comment:
General comment concerning the code base is to not use these unnecessary abbreviations

Reply:
I removed the worst offenders I noticed in here, for now